source/extensions/common/wasm/null/null.cc

```diff
@@ -18,7 +18,7 @@ namespace Common {
 namespace Wasm {
 namespace Null {
 
-WasmVmPtr createVm() { return std::make_unique<NullVm>(); }
+WasmVmPtr createVm(Stats::ScopeSharedPtr scope) { return std::make_unique<NullVm>(scope); }
```

Comment:
nit: `std::move(scope)` or make param const ref (same for places below)

Reply:
I was wondering about how effective copy elision had gotten so I ran a simple test:

struct Foo {
  Foo(int x) : x_(x) {}
  int x_;
};

struct Bar {
  Bar(std::shared_ptr<Foo> x) : x_(std::move(x)) {}
  std::shared_ptr<Foo> x_;
};

int main() {
  auto x = std::make_shared<Foo>(1);
  Bar b(std::move(x));
  printf("%d\n", b.x_->x_);
}

************ code 1 *******************

struct Foo {
  Foo(int x) : x_(x) {}
  int x_;
};

struct Bar {
  Bar(std::shared_ptr<Foo> x) : x_(x) {}
  std::shared_ptr<Foo> x_;
};

int main() {
  auto x = std::make_shared<Foo>(1);
  Bar b(x);
  printf("%d\n", b.x_->x_);
}

******************* code 2 *********************

These two codes return the exact same file when compiled:

clang++ (version 9) -O3 -S

Reply:
Half this change can be upstreamed, but the wasm.cc part can't be (yet).

I will push the upstream of the applicable part once this goes in.

Reply:
either is fine, but I guess clang-tidy may flag you there: https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html